internal/storage/bucket_handle.go

```diff
@@ -27,24 +27,55 @@ import (
 	"net/http"
 
 	"cloud.google.com/go/storage"
+	control "cloud.google.com/go/storage/control/apiv2"
+	"cloud.google.com/go/storage/control/apiv2/controlpb"
+	"github.com/googleapis/gax-go/v2"
+	"github.com/googlecloudplatform/gcsfuse/v2/internal/logger"
 	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/gcs"
 	"github.com/googlecloudplatform/gcsfuse/v2/internal/storage/storageutil"
+
 	"google.golang.org/api/googleapi"
 	"google.golang.org/api/iterator"
 )
 
 type bucketHandle struct {
 	gcs.Bucket
-	bucket     *storage.BucketHandle
-	bucketName string
-	bucketType gcs.BucketType
+	bucket        *storage.BucketHandle
+	bucketName    string
+	bucketType    gcs.BucketType
+	controlClient StorageControlClientInterface
 }
 
 func (bh *bucketHandle) Name() string {
 	return bh.bucketName
 }
 
 func (bh *bucketHandle) BucketType() gcs.BucketType {
```

Comment:
Can we please write a comment here, similar to below:
```During the first BucketType() call, type is inferred by making a GCS call and cached in a local variable. So, that the first call will be a little slower, and further call will be faster since no GCS call.

Also, this is thread safe, but parallelly calling by mutiple threads the first time might result in the more than one GCS call. So, it's good to call it while mounting first to avoid this case.
```




(